### PR TITLE
fix(deploy-script-support): RESM compatible bundle powers

### PR DESCRIPTION
**Pull request by mhofman**
refs: https://github.com/Agoric/testnet-load-generator/pull/85
refs: #6826 

## Description

This makes the `deploy-script-support` RESM compatible as needed for the loadgen which hasn't converted for backwards compatibility.

### Security Considerations

None

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

Manually ran the updated loadgen, against this branch

### CODE REVIEW

File: packages/agoric-cli/src/deploy.js [hunk 1]
```diff
@@ -326,7 +326,10 @@ export default async function deployMain(progname, rawArgs, powers, opts) {
             const fileName = paths.pop();
             try {
               return require.resolve(fileName, {
-                paths: [...paths, path.dirname(moduleFile)],
+                paths: [
+                  path.resolve(path.dirname(moduleFile), ...paths),
```
Comment (mhofman) [inline packages/agoric-cli/src/deploy.js:330]: This is more of a drive-by, as I believe the spread was a mistake: for `path.resolve`, `paths` components would be joined if not absolute, where for `require.resolve`, each `paths` is a new root. Since I believe `dirname(moduleFile)` is absolute, this should result in the correct behavior and never use `cwd`.

File: patches/@endo+static-module-record+0.7.15.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/@endo/static-module-record/src/transformSource.js b/node_modules/@endo/static-module-record/src/transformSource.js
+index cba7f8e..3a814d9 100644
+--- a/node_modules/@endo/static-module-record/src/transformSource.js
++++ b/node_modules/@endo/static-module-record/src/transformSource.js
+@@ -1,7 +1,7 @@
+ import * as babelParser from '@babel/parser';
+ import babelGenerate from '@agoric/babel-generator';
+ import babelTraverse from '@babel/traverse';
+-import babelTypes from '@babel/types';
++import * as babelTypes from '@babel/types';
```
Comment (mhofman) [inline patches/@endo+static-module-record+0.7.15.patch:10]: See https://github.com/endojs/endo/pull/1466

File: packages/agoric-cli/src/deploy.js (hunk 1 above)
Comment (mhofman) [reply packages/agoric-cli/src/deploy.js:330]: This was introduced in https://github.com/Agoric/agoric-sdk/pull/5132/commits/d83dc644d91c50b2ab5bbcb45c547956db9fec3f and I didn't see a reasoning behind the spread.

Comment (michaelfig) [reply packages/agoric-cli/src/deploy.js:330]: Thanks for catching this!

**Review by michaelfig — APPROVED** [review]
Thanks for freshening this bitrot. :)
